pkg/volume/csi/csi_attacher.go

```diff
@@ -365,6 +366,16 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		mountOptions = spec.PersistentVolume.Spec.MountOptions
 	}
 
+	volAttribs := csiSource.VolumeAttributes
+	// Inject pod information into volume_attributes
+	podInfoEnabled, err := c.plugin.podInfoEnabled(string(csiSource.Driver))
+	if err != nil {
+		return volumetypes.NewTransientOperationFailure(log("attacher.MountDevice failed to assemble volume attributes: %v", err))
+	}
+	if podInfoEnabled {
+		volAttribs = mergeMap(volAttribs, getPodInfoAttrs(c.pod, storage.VolumeLifecyclePersistent))
```

Comment:
I think some of the attributes here may not make sense as NodeStage is once per volume, but NodePublish is once per pod.

Namespace could be something we can share across pods and multiple Pods sharing the same PVC will be in the namespace. It is possible for someone to create multiple PVs using the same volumeHandle and those PVCs could be in different namespaces. Would that cause issues for your driver?

Reply:
In addition to what Michelle wrote, one in-line in-tree Azure File volume can be used by several pods in multiple namespaces. Is it possible that such volume needs namespace in NodeStage when it's migrated to CSI? It's not possible to reliably provide one.